pwnlib/libcdb.py

```diff
@@ -645,10 +744,74 @@ def search_by_sha256(hex_encoded_id, unstrip=True):
         >>> hex(ELF(filename).symbols.read)
         '0xeef40'
     """
-    return search_by_hash(hex_encoded_id, 'sha256', unstrip)
+    return search_by_hash(hex_encoded_id, 'sha256', unstrip, offline)
+
+def get_libc_symbols(path):
+    """
+    Given symbol file path, the content format is as follows:
+        ...
+        puts 000xxxxx
+        read 0000xxxx
+        write 0000xxxx
+        ...
+
+    Arguments:
+        path(str):
+            symbol file path.
+
+    Retruns:
+        a dictionary of the form: { name : address }, or :const:`{}`
+    """
 
+    syms = {}
 
+    with open(path, "r") as fd:
+        for x in fd.read().strip().split("\n"):
+            name, addr = x.split(" ")
+            syms[name] = int(addr, 16)
 
+    return syms
+
+def get_libc_info(db_path, libs_id, syms, libc_path=None, symbols_path=None, libs_url=None):
+    info = {}
+
+    info["id"] = libs_id
+    info["libc_path"] = os.path.join(db_path, f"{libs_id}.so") if not libc_path else libc_path
+    info["libs_url"] = read(os.path.join(db_path, f"{libs_id}.url")).decode().strip() if not libs_url else libs_url
+    info["symbols_path"] = os.path.join(db_path, f"{libs_id}.symbols") if not symbols_path else symbols_path
+
+    info["buildid"] = get_build_id(info["libc_path"])
+    info["md5"] = md5filehex(info["libc_path"])
+    info["sha1"] = sha1filehex(info["libc_path"])
+    info["sha256"] = sha256filehex(info["libc_path"])
+
+    info["symbols"] = {}
+    info["symbols"]["__libc_start_main_ret"] = hex(syms["__libc_start_main_ret"])
+    info["symbols"]["dup2"] = hex(syms["dup2"])
+    info["symbols"]["open"] = hex(syms["open"])
+    info["symbols"]["printf"] = hex(syms["printf"])
+    info["symbols"]["puts"] = hex(syms["puts"])
+    info["symbols"]["read"] = hex(syms["read"])
+    info["symbols"]["str_bin_sh"] = hex(syms["str_bin_sh"])
+    info["symbols"]["system"] = hex(syms["system"])
+
+    return info
+
+def get_build_id(path):
```

Comment:
I believe [ELF](https://docs.pwntools.com/en/latest/elf/elf.html#pwnlib.elf.elf.ELF) loading encompasses too many components, and this might impact the runtime performance when traversing the local libc database. As a result, I've extracted the crucial logic into a `get_build_id` function.

I haven't conducted tests in this regard, my understanding is primarily based on the code itself.